### PR TITLE
[INFRA] Run travis as nightly on multiple CPU archs

### PR DESCRIPTION
**Pull request by eseiler**
Resolves https://github.com/seqan/seqan3/issues/1482

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -20,107 +20,31 @@ addons:
...
+  - BUILD=header      CXX=g++-7  CC=gcc-7  BUILD_TYPE=Release
+  - BUILD=snippet     CXX=g++-7  CC=gcc-7  BUILD_TYPE=Release
 
 install:
-  - |
-    HOME_BIN_PATH=`realpath ~/bin`
-    mkdir -p ${HOME_BIN_PATH}
-  - |
-    # install cmake
-    CMAKE_VERSION="3.7.2"
```
Comment (marehr) [inline .travis.yml:109]: If we don't fix the cmake version, the log will get to big, we need to fix our cmake script to use the less verbose version.

```cmake
    if (NOT CMAKE_VERSION VERSION_LESS 3.10) # cmake >= 3.10
        gtest_discover_tests(${target} TEST_PREFIX "${test_name}::" PROPERTIES TIMEOUT "30")
    else ()
        add_test (NAME "${test_name}" COMMAND ${target})
    endif ()
```

should be

```cmake
    option (SEQAN3_VERBOSE_TESTS ON)
    if (SEQAN3_TEST_VERBOSE_TESTS AND NOT CMAKE_VERSION VERSION_LESS 3.10) # cmake >= 3.10
        gtest_discover_tests(${target} TEST_PREFIX "${test_name}::" PROPERTIES TIMEOUT "30")
    else ()
        add_test (NAME "${test_name}" COMMAND ${target})
    endif ()
```

**Review by marehr — COMMENTED** [review]
looks good, we need to make the test output shorter for travis

File: .travis.yml (hunk 1 above)
Comment (eseiler) [reply .travis.yml:109]: That means we should run the nightlies with `-DSEQAN3_VERBOSE_TESTS=OFF` ?

Comment (marehr) [reply .travis.yml:109]: jop

**Comment by eseiler** [comment]
I will do the adjustements to the cmake script in the release branch (CI should also use SEQAN3_VERBOSE_TESTS=OFF)

**Review by marehr — APPROVED** [review]
lgtm!

File: .travis.yml [hunk 2]
```diff
@@ -20,107 +20,36 @@ addons:
...
       - g++-10
       - lcov
+      - cmake
+arch:
+  - arm64
+  - s390x
+  - ppc64le
+
+env:
+  - BUILD=coverage    CXX=g++-7  CC=gcc-7  BUILD_TYPE=Debug
```
Comment (rrahn) [inline .travis.yml:30]: this is neat 🤓

File: .travis.yml [hunk 3]
```diff
@@ -138,10 +67,14 @@ before_script:
 
 script:
   - export SEQAN3_NO_VERSION_CHECK=1
+  - export CCACHE_COMPRESS=true
+  - export CCACHE_COMPRESSLEVEL=6
+  - export CCACHE_MAXSIZE=5G
```
Comment (rrahn) [inline .travis.yml:72]: this is now enough or not important because it is a nightly build?

File: .travis.yml [hunk 4]
```diff
@@ -138,10 +67,14 @@ before_script:
...
   - make -k -j2
   - |
-    if test coverage != "${BUILD}"; then
-      if [[ "${BUILD}" =~ ^(snippet)$ ]]; then
+    if [[ "${{ matrix.build }}" =~ ^(coverage)$ ]]; then
+      : ; else
```
Comment (rrahn) [inline .travis.yml:76]: what happens here?

**Review by rrahn — COMMENTED** [review]
looks great. Just some minor questions.

File: .travis.yml (hunk 2 above)
Comment (eseiler) [reply .travis.yml:30]: Yes, but unfortunately you cannot name it - naming only works for `include`.

But when you use include, no matrix is automatically generated (job X architecture).

File: .travis.yml (hunk 4 above)
Comment (eseiler) [reply .travis.yml:76]: We don't need to run ctest if we are building coverage - running the tests is already part of building the coverage, so I just do nothing (no-op operator `:`).

Comment (eseiler) [reply .travis.yml:76]: It was `if test coverage != "${BUILD}"; then` before, but I didn't understand what it did until I removed it; I think not mixing different syntaxes probably helps understanding

File: .travis.yml (hunk 3 above)
Comment (eseiler) [reply .travis.yml:72]: When we compress it, it gets considerably smaller, I think the GitHub actions ccaches are around 30-60MiB, so we should have always compressed them; 5G are now more than enough

**Review by rrahn — APPROVED** [review]
thank you

**Pull request merged**